Snowpack

JoviDeCroock opened the pull request:
Depends on: https://github.com/pikapkg/snowpack/pull/332 

Just like `vite` at this moment it's lacking support for HMR on hooks due to the AST-part missing. I'll try and see if we can add this soon'ish.

To test this locally I added `react-refresh/babel` to my `babel.config.json`

FredKSchott commented:
Looks good! This API just went out with `snowpack@2.0.0-beta.24` (`npm install snowpack@next`). Let me know if I can help at all with the integration.

FredKSchott commented on packages/snowpack/README.md:
````diff
@@ -0,0 +1,21 @@
+
+```json
+{
+  "plugins": ["@prefresh/snowpacks"]
````
Typo :)

FredKSchott commented on packages/snowpack/src/index.js:
```diff
@@ -0,0 +1,61 @@
+                }
+                $CurrentModule$ = module;
+              } catch(e) {
+                window.location.reload();
```
Would it help to replace this with an import.meta.hot.reject() call instead? Feels lame to force you to reload yourself.

JoviDeCroock replied:
Hmm, recently I discovered [this one](https://webpack.js.org/api/hot-module-replacement/#invalidate) which to me felt likea good course of action.

FredKSchott replied:
Nice, +1 for matching their terminology.

JoviDeCroock commented:
@FredKSchott I don't know if this is an issue on how I'm using it but I'm not seeing `babel` run atm initially I just added it to the `preact-scripts` found in the starter for `Preact`.

This didn't work so I made a local `babel.config.json` and added `"plugin:js,jsx": "@snowpack/plugin-babel"` to my package.json. My logs in the babel-plugin don't seem to trigger.

FredKSchott commented:
Ah yea, we just finalized the interface:

- we use esbuild internally so no Babel needed, works with Preact & JSX, you can see the new preact CSA template to see that in action
- I can see how thats not obvious tho, we should show that that’s the new default in our dev output
- “plugin:js,jsx” is now just “build:js,jsx” ... and it will be connected automatically for you by just putting the plugin in your “plugins” config
- I think I updated the docs on snowpack.dev with these updates but it’s still new so still in progress

JoviDeCroock commented:
Merging while we work on the bubbling HMR api for snowpack!
